packages/compass-collection/src/modules/collection-tab.ts

```diff
@@ -22,32 +31,82 @@ type CollectionThunkAction<R, A extends AnyAction = AnyAction> = ThunkAction<
     dataService: DataService;
     workspaces: ReturnType<typeof workspacesServiceLocator>;
     experimentationServices: ReturnType<typeof experimentationServiceLocator>;
+    logger: Logger;
+    preferences: PreferencesAccess;
   },
   A
 >;
 
+export enum SchemaAnalysisStatus {
```

Comment:
fyi there's another thread that advocates compass's preference for unions over enums in https://github.com/mongodb-js/compass/pull/7181#discussion_r2264145968.